include/CUDACopy.hpp

```diff
@@ -43,20 +43,15 @@ void copy(fg::Image& out, const T * devicePtr)
 }
 
 /*
- * Below functions takes any renderable forge object that has following member functions
- * defined
- *
- * `unsigned Renderable::vbo() const;`
- * `unsigned Renderable::size() const;`
- *
- * Currently fg::Plot, fg::Histogram objects in Forge library fit the bill
+ * Below functions expects OpenGL resource Id and size in bytes to copy the data from
+ * CUDA device memory location to graphics memory
  */
-template<class Renderable, typename T>
-void copy(Renderable& out, const T * devicePtr)
+template<typename T>
+void copy(const int resourceId, const T * devicePtr)
 {
     cudaGraphicsResource *cudaVBOResource;
-    CUDA_ERROR_CHECK(cudaGraphicsGLRegisterBuffer(&cudaVBOResource, out.vbo(), cudaGraphicsMapFlagsWriteDiscard));
-
+    CUDA_ERROR_CHECK(cudaGraphicsGLRegisterBuffer(&cudaVBOResource, resourceId,
```

Comment:
Rename this macro to be `FORGE_CUDA_CHECK` or something along those lines so it won't conflict with user defined macros.